src/Corporation/Actions.ts

```diff
@@ -251,7 +251,7 @@ function assertSmartSupplyOption(value: string): asserts value is CorpSmartSuppl
   }
 }
 
-export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: string): void {
+export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: CorpSmartSupplyOption): void {
   assertSmartSupplyOption(useOption);
```

Comment:
This also should either take in a string and perform validation itself, or it should take in a valid type and not perform its own validation.